src/Microsoft.AspNet.Razor.Runtime/Runtime/TagHelpers/TagHelperRunner.cs

```diff
@@ -31,9 +32,10 @@ public class TagHelperRunner
                 executionContext.AllAttributes,
                 executionContext.Items,
                 executionContext.UniqueId);
-            var orderedTagHelpers = executionContext.TagHelpers.OrderBy(tagHelper => tagHelper.Order);
 
-            foreach (var tagHelper in orderedTagHelpers)
+            OrderTagHelpers(executionContext.TagHelpers);
```

Comment:
While you're here, these should be `for` loops if they happen at runtime with an `IList` or `IReadOnlyList`.